ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/netty/NettyP2PNetwork.java

```diff
@@ -497,6 +546,42 @@ public void start() {
     };
   }
 
+  private synchronized void handleBlockAddedEvent(
+      final BlockAddedEvent event, final Blockchain blockchain) {
+    connections
+        .getPeerConnections()
+        .forEach(
+            peerConnection -> {
+              if (!isPeerConnectionAllowed(peerConnection)) {
+                peerConnection.disconnect(DisconnectReason.REQUESTED);
+              }
+            });
+  }
+
+  private boolean isPeerConnectionAllowed(final PeerConnection peerConnection) {
+    LOG.trace(
+        "Checking if connection with peer {} is permitted", peerConnection.getPeer().getNodeId());
+
+    final String localNodeId = ourPeerInfo.getNodeId().toString().substring(2);
```

Comment:
refactor the local & remote enode bits to a seperate function like `peerToEnode(PeerInfo peer, InetSocketAddress address)`